File: src/page/JBTN.java
package page;

import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.util.function.Consumer;

import javax.swing.JButton;

public class JBTN extends JButton implements LocComp {

	private static final long serialVersionUID = 1L;

	private final LocSubComp lsc;

	public JBTN() {
		lsc = new LocSubComp(this);
	}

	public JBTN(int i, String str) {
		this();
		lsc.init(i, str);
	}

	public JBTN(String str) {
		this(-1, str);
	}

	@Override
	public LocSubComp getLSC() {
		return lsc;
	}

	public void setLnr(Consumer<ActionEvent> c) {
		addActionListener(new ActionListener() {

			@Override
			public void actionPerformed(ActionEvent e) {
				c.accept(e);
			}

		});
	}

}

File: src/page/JBTN.kt
package page

import common.util.lang.LocaleCenter
import page.MainFrame
import java.awt.event.ActionEvent
import java.awt.event.ActionListener
import java.util.function.Consumer
import java.util.function.Supplier
import javax.swing.JButton

class JBTN() : JButton(), LocComp {
    private val lsc: LocSubComp

    constructor(binder: LocaleCenter.Binder?) : this() {
        lsc.init(binder)
    }

    constructor(i: Int, str: String) : this() {
        lsc.init(i, str)
    }

    constructor(str: String) : this(-1, str) {}

    override fun getLSC(): LocSubComp {
        return lsc
    }

    fun setLnr(c: Consumer<ActionEvent?>) {
        addActionListener(object : ActionListener {
            override fun actionPerformed(e: ActionEvent?) {
                c.accept(e)
            }
        })
    }

    fun setLnr(s: Supplier<Page?>) {
        setLnr { e: ActionEvent? -> MainFrame.Companion.changePanel(s.get()) }
    }

    companion object {
        private const val serialVersionUID = 1L
    }

    init {
        lsc = LocSubComp(this)
    }
}
